cmd/sops/main.go

```diff
@@ -1385,6 +1385,10 @@ func main() {
 					Name:  "output-type",
 					Usage: "currently json, yaml, dotenv and binary are supported. If not set, sops will use the input file's extension to determine the output format",
 				},
+				cli.BoolFlag{
+					Name:  "value-file",
+					Usage: "treat 'value' as a file to read the actual value from (avoids leaking secrets in process listings)",
+				},
```

Comment:
How about adding a (mutually exclusive) further option that reads from stdin instead of a file? That would provide a platform independent way of reading from stdin (`/dev/stdin` doesn't work on Windows).

Reply:
I personally think it's better to make things explicit, and let the user explicitly ask for reading from stdin, instead of implicitly just doing that and confusing users which forget the value argument and then wonder why the program is "hanging".

Reply:
I added "treat - as stdin", and rebased.

Reply:
Sorry for the late reply; I think having a dedicated option is better than using `-`. (We had a similar discussion in https://github.com/getsops/sops/issues/739#issuecomment-2343790721; after reading https://github.com/FiloSottile/age/issues/494 I prefer a separate option. Magic can always lead to trouble, including security problems, and treating `-` different from other files would be some magic.)